sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -70,6 +71,12 @@ struct mc_private {
 	struct snd_soc_jack sdw_headset;
 };
 
+struct sdw_dai_link {
+	int dai_id;
+	int num_adr;
+	u64 *adr;
+};
```

Comment:
why do you need this, you already have all the information in mach_params? This looks like a lot of changes only to group the devices by dai_id?
it might be simpler to add a new field in the ACPI tables to say which devices are logically grouped